go/vt/vtgate/planbuilder/operators/horizon_planning.go

```diff
@@ -66,16 +66,10 @@ func planHorizon(ctx *plancontext.PlanningContext, in *Horizon) (ops.Operator, r
 	needsOrdering := len(qp.OrderExprs) > 0
 	canShortcut := isRoute && sel.Having == nil && !needsOrdering
 
-	// If we still have a HAVING clause, it's because it could not be pushed to the WHERE,
-	// so it probably has aggregations
-	switch {
-	case qp.NeedsAggregation() || sel.Having != nil:
-		return nil, rewrite.VisitChildren, errNotHorizonPlanned
-	case canShortcut && !needsOrdering && !qp.NeedsDistinct() && in.Select.GetLimit() == nil:
+	if !qp.NeedsAggregation() && sel.Having == nil && canShortcut && !needsOrdering && !qp.NeedsDistinct() && in.Select.GetLimit() == nil {
```

Comment:
`canShortcut` is only used in the if condition which can be expanded as it will only add isRoute check rest of the checks are already present.